src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupCodec.java

```diff
@@ -0,0 +1,166 @@
+package org.broadinstitute.hellbender.utils.codecs.sampileup;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.SAMUtils;
+import htsjdk.tribble.AsciiFeatureCodec;
+import htsjdk.tribble.Feature;
+import htsjdk.tribble.SimpleFeature;
+import htsjdk.tribble.exception.CodecLineParsingException;
+import htsjdk.tribble.readers.LineIterator;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Decoder for single sample SAM pileup data.
+ *
+ * See the <a href="http://samtools.sourceforge.net/pileup.shtml">Pileup format documentation</a> for more details
+ * on the format
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class SAMPileupCodec extends AsciiFeatureCodec<SAMPileupFeature> {
+
+    // including also space as delimiter
+    private static final Pattern SPLIT_PATTERN = Pattern.compile("\\t|( +)");
+
+    // although the minimum number of fields is 6 (see specifications), if the coverage is 0 sometimes bases and qualities does not appear
+    private static int MINIMUM_FIELDS = 4;
+    // number of maximum fields expected
+    private static final int MAXIMUM_FIELDS = 6;
+
+    // regular expression for indels
+    private static final Pattern INDEL_REGEXP = Pattern.compile("([0-9]+).*");
+    // codec file extensions
+    public static final List<String> SAM_PILEUP_FILE_EXTENSIONS = Arrays.asList("pileup", "mpileup", "samp");
+
+    public SAMPileupCodec() {
+        super(SAMPileupFeature.class);
+    }
+
+    @Override
+    public Object readActualHeader(final LineIterator lineIterator) {
+        // No header for this format
+        return null;
+    }
+
+    /**
+     * Only files with {@link #SAM_PILEUP_FILE_EXTENSIONS} could be parsed
+     * @param path path the file to test for parsability with this codec
+     * @return {@code true} if the path has the correct file extension {@link #SAM_PILEUP_FILE_EXTENSIONS}, {@code false} otherwise
+     */
+    @Override
+    public boolean canDecode(final String path) {
+        return SAM_PILEUP_FILE_EXTENSIONS.stream().anyMatch(ext -> path.endsWith("."+ext));
+    }
+
+    public SAMPileupFeature decode(String line) {
+        // Split the line
+        final String[] tokens = SPLIT_PATTERN.split(line.trim(), -1);
+        // check the number of fields
+        if(tokens.length < MINIMUM_FIELDS || tokens.length > MAXIMUM_FIELDS) {
+            throw new CodecLineParsingException("The SAM pileup line didn't have the expected number of columns. " +
```

Comment:
Could you include the actual text of the line in this comment as well as the number of columns and the expected range?